### PR TITLE
Convert this addon to v2 format

### PR DESCRIPTION
**Pull request by nlfurniss**
## Summary
Converts this addont to addon v2 format by following [this guide](https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md).

Done in two commits
- Part 1: Separate Addon from Dummy App
- Addon V2

Also bumped the node version since this’ll be a major bump anyway.

## Testing Done
`addon` builds and `ember test/serve/build` all work correctly and all tests pass

### CODE REVIEW

**Comment by nlfurniss** [comment]
CI doesn't run for any PR; is it because [of this](https://github.com/ember-mocks/ember-date-service/blob/master/.github/workflows/ci-build.yml#L8)?

Should it be?

```diff
-  pull_request: {}
+  pull_request:

```

**Comment by scalvert** [comment]
> CI doesn't run for any PR; is it because [of this](https://github.com/ember-mocks/ember-date-service/blob/master/.github/workflows/ci-build.yml?rgh-link-date=2022-05-20T04%3A43%3A17Z#L8)?
> 
> Should it be?
> 
> ```diff
> -  pull_request: {}
> +  pull_request:
> ```

No, that should allow all PRs to run CI. GitHub disables workflows for repos that don't have a lot of activity. The workflow was disabled, so I re-enabled it.

**Comment by scalvert** [comment]
It looks like we're going to have to bump node along with this change. Likely to Node 14.

File: test-app/ember-cli-build.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+const EmberApp = require('ember-cli/lib/broccoli/ember-app');
```
Comment (nlfurniss) [inline test-app/ember-cli-build.js:3]: Not sure how to fix this, as this dep is defined in the [test app's package.json](https://github.com/ember-mocks/ember-date-service/pull/174/files#diff-b03f4acb30e04ceb12a40763ef649759b3f632df2357bcec14fab6dd29ebbfb4R28), but doesn't get installed in the `test-app/node_modules`

**Comment by nlfurniss** [comment]
All tests pass locally now:

```
All 7 scenarios succeeded
✨  Done in 278.00s.
```

**Comment by nlfurniss** [comment]
Running `cd addon && yarn build`  adds this line to the the addon package.json file:

```json
    "app-js": {
      "./services/date.js": "./dist/_app_/services/date.js"
    }
```

Which is required for tests to pass locally, but seems to be why they're failing in CI

**Comment by scalvert** [comment]
So any reason why we're not converting to a monorepo? This is the [recommended structure of v2 addons](https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md#monorepo-organization), and is what we used for ember-lifeline's conversion (which worked out really well).

**Comment by scalvert** [comment]
Nevermind, I see you are following those instructions (which don't use `packages/*` as the defined workspaces).

File: .gitignore [hunk 2]
```diff
@@ -23,3 +23,5 @@
 /.node_modules.ember-try/
 /bower.json.ember-try
 /package.json.ember-try
+
+.DS_Store
```
Comment (scalvert) [inline .gitignore]: I think this should be in your global gitignore rather than the repo-specific one.

File: addon/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  root: true,
+  parser: 'babel-eslint',
```
Comment (scalvert) [inline addon/.eslintrc.js]: This parser is deprecated, so should probably be updated too. It's OK to do this in a separate PR.

File: addon/.gitignore [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+/npm-debug.log*
+/testem.log
+/yarn-error.log
+
+# ember-try
+/.node_modules.ember-try/
+/bower.json.ember-try
+/package.json.ember-try
+
+.DS_Store
```
Comment (scalvert) [inline addon/.gitignore]: Probably should be in your global gitignore.

**Review by scalvert — CHANGES_REQUESTED** [review]

File: test-app/package.json [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    "git": {
+      "tagName": "v${version}",
+      "changelog": "git log --pretty=format:\"* %s (%h)\" --no-merges --perl-regexp --author=\"^((?!dependabot-preview).*)$\" ${latestTag}...HEAD"
+    },
+    "github": {
+      "release": true,
+      "tokenRef": "GITHUB_AUTH"
+    }
+  },
```
Comment (scalvert) [inline test-app/package.json]: We're never going to release this package, so probably don't need this release-it config?

**Review by scalvert — APPROVED** [review]

**Comment by scalvert** [comment]
Thank you so so so much @nlfurniss for doing this work. It's massively appreciated!

**Pull request merged**